### ISSUE TITLE
[Platform Compatibility Analyzer] Type-level and member-Level attributes do not override assembly-level attributes

### ISSUE THREAD

**Issue by jeffhandley** [dotnet/roslyn-analyzers#4020]
### Describe the bug

When a `[SupportedOSPlatform]` attribute is applied to an assembly, `[UnsupportedOSPlatform]` attributes on types and members do not override the assembly attribute.

### Steps To Reproduce

```csharp
using System.Runtime.Versioning;

[assembly:SupportedOSPlatform("browser")]

namespace PlatformCompatDemo
{
    static class Program
    {
        public static void Main()
        {
            CrossPlatformApis.DoesNotWorkOnBrowser(); // Diagnostic expected; none occurs with assembly attribute
            var nonBrowser = new NonBrowserApis();    // Diagnostic expected; none occurs with assembly attribute
        }
    }

    public class CrossPlatformApis
    {
        [UnsupportedOSPlatform("browser")]
        public static void DoesNotWorkOnBrowser() { }
    }

    [UnsupportedOSPlatform("browser")]
    public class NonBrowserApis
    {
    }
}
```
### Expected behavior

* Diagnostic when calling `CrossPlatformApis.DoesNotWorkOnBrowser();
* Diagnostic when calling `new NonBrowserApis();`

### Actual behavior

* No diagnostics

### Analyzer

Platform Compatibility Analyzer; local build from https://github.com/dotnet/roslyn-analyzers/pull/3917/commits/1eba4b85b49c296650db237a417004ed37694f66

**Comment by jeffhandley** [comment]
This is not considered as a blocker for RC1 because this scenario will not exist yet within the .NET libraries themselves, but it could exist within a customer's scenario where:

1. An assembly is marked with a set of supported platform
2. Some APIs are marked as unsupported on some of those platforms

This scenario might surface when we start producing assemblies for `ios`, `tvos`, and `watchos` where there will likely be some methods within those assemblies that don't have support on all of those platforms; it's easy to imagine an assembly that generally works across all those platforms but doesn't work on `watchos`.

So we won't block the RC1 merge for this, but we need to define more test cases around composition of attributes across multiple levels, define what the behavior is for all possible combinations, and review those behaviors to see what should be updated.  Depending on those findings, we may need to make changes before the .NET 5 final release.

**Comment by buyaa-n** [comment]
Fixed with https://github.com/dotnet/roslyn-analyzers/pull/3917/commits/99883854e87fce50449674b9638ece153becd082
